### ISSUE TITLE
Error while trying to authenticate user

### ISSUE THREAD

**Issue by holiiveira** [grafana/grafana#28608]
**What happened**:
Error when logging in with user:

```
t=2020-10-28T07:33:13-0300 lvl=eror msg="Error while trying to authenticate user" logger=context userId=0 orgId=0 uname= error="Error 1406: Data too long for column 'auth_id' at row 1" remote_addr=172.17.xx.xxx
t=2020-10-28T07:33:13-0300 lvl=eror msg="Request Completed" logger=context userId=0 orgId=0 uname= method=POST path=/login status=500 remote_addr=172.17.xx.xxx time_ms=33 size=53 referer=https://noc.xxxx.com.br/grafana/login
```

**Anything else we need to know?**:
Problem occurred after database migration, and migration Grafana version: 7.2.0 to 7.2.2.

**before**: 10.1.45-MariaDB-0+deb9u1 Debian 9.12
**after**: 8.0.22 MySQL Community Server - GPL

```
mysql> desc user_auth;
+----------------------+--------------+------+-----+---------+----------------+
| Field                | Type         | Null | Key | Default | Extra          |
+----------------------+--------------+------+-----+---------+----------------+
| id                   | bigint       | NO   | PRI | NULL    | auto_increment |
| user_id              | bigint       | NO   | MUL | NULL    |                |
| auth_module          | varchar(190) | NO   | MUL | NULL    |                |
| auth_id              | varchar(190) | YES  |     | NULL    |                |
| created              | datetime     | NO   |     | NULL    |                |
| o_auth_access_token  | text         | YES  |     | NULL    |                |
| o_auth_refresh_token | text         | YES  |     | NULL    |                |
| o_auth_token_type    | text         | YES  |     | NULL    |                |
| o_auth_expiry        | datetime     | YES  |     | NULL    |                |
+----------------------+--------------+------+-----+---------+----------------+
```

I have users where the auth_id field exceeds 190 characters. However, this error did not occur before migrating the database. 

**Environment**:
- Grafana version: v7.2.2 (ad9d408ac2)

**Comment by gpxlnx** [comment]
I am also facing the same problem described above.

**Comment by marefr** [comment]
It was 255 earlier, but we had to change it to 190 via #11867 because certain mysql versions don't support having indices with a greater varchar length than 190. I'm afraid there's not much more we can do here.

If your mysql supports having indices with a greater varchar length than 190 I would suggest you to try and increase that using

```sql
ALTER TABLE user_auth MODIFY auth_id VARCHAR(<length>);
```

**Comment by marefr** [comment]
/duplicate of #12651 

Didn't see that until now.

**Comment by svestenik** [comment]
Issue still present with Grafana 8.3.4 and MARIADB 10.3.28. After migrating backend from sqllite to mysql LDAP stopped working with error "lvl=eror msg="Error while trying to authenticate user" logger=context userId=0 orgId=0 uname= error="Error 1406: Data too long for column 'auth_id' at row 1""
